STUDIO/org.openl.rules.webstudio/src/org/openl/rules/rest/MailService.java

```diff
@@ -0,0 +1,169 @@
+        this.userSettingManagementService = userSettingManagementService;
+        this.userManagementService = userManagementService;
+        this.currentUserInfo = currentUserInfo;
+        this.propertyResolver = propertyResolver;
+        this.validationProvider = validationProvider;
+    }
+
+    @GET
+    @Path("/verify/{token}")
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response verify(@PathParam("token") String token) {
+        String username = currentUserInfo.getUserName();
+        User user = userManagementService.getUser(username);
+        String dbToken = userSettingManagementService.getStringProperty(username, MAIL_VERIFY_TOKEN);
+        if (Objects.equals(dbToken, token) && Objects.nonNull(user)) {
+            userManagementService.updateUserData(user.getUsername(),
+                user.getFirstName(),
+                user.getLastName(),
+                null,
+                false,
+                user.getEmail(),
+                user.getDisplayName(),
+                true);
+            userSettingManagementService.setProperty(username, MAIL_VERIFY_TOKEN, "");
+            updateCurrentUserEmailVerified();
+            return Response.noContent().build();
+        } else {
+            throw new BadRequestException("mail.wrong.token");
+        }
+    }
+
+    private void updateCurrentUserEmailVerified() {
+        Optional.ofNullable(currentUserInfo.getAuthentication()).map(authentication -> {
+            if (authentication.getPrincipal() instanceof SimpleUser) {
+                return (SimpleUser) authentication.getPrincipal();
+            } else if (authentication.getDetails() instanceof SimpleUser) {
+                return (SimpleUser) authentication.getDetails();
+            } else {
+                return null;
+            }
+        })
+            .ifPresent(
+                simpleUser -> simpleUser.setExternalFlags(UserExternalFlags.builder(simpleUser.getExternalFlags())
+                    .withFeature(UserExternalFlags.Feature.EMAIL_VERIFIED)
+                    .build()));
+    }
+
+    @POST
+    @Path("/send/{username}")
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response sendVerification(@Context UriInfo uriInfo, @PathParam("username") String username) {
+        if (!currentUserInfo.getUserName().equals(username)) {
+            SecurityChecker.allow(Privileges.ADMIN);
+        }
+        User user = userManagementService.getUser(username);
+        String token = RandomStringUtils.random(8, false, true);
+        String email = user.getEmail();
+        boolean emailWasSent = mailSender.sendVerification(email, createVerificationLink(uriInfo, token));
+        if (emailWasSent) {
+            userSettingManagementService.setProperty(username, MAIL_VERIFY_TOKEN, token);
+            return Response.ok("Sent to " + email).build();
+        } else {
+            return Response.status(Response.Status.FORBIDDEN).build();
+        }
+    }
+
+    @GET
+    @Path("/config")

```

Comment:
``/settings`` path is already chosen in ManagementService. Please use the same to keep it more unified.